### ISSUE TITLE
Controlled Generation (response_schema) missing

### ISSUE THREAD

**Issue by diervo** [googleapis/nodejs-vertexai#392]
Per the documentation in other SDKs and some Community members' [blog posts](https://medium.com/google-cloud/vertex-ai-controlled-generation-with-gemini-8a60d0581f62), `response_schema` seems that it should be supported. However, it is not part of the typescript definition.

So is this by design, or is just a missing type in the `GenerationConfig` type definition?

Seems that maybe should be a follow-up to https://github.com/googleapis/nodejs-vertexai/commit/93f6d70660dbee44f81766544047da7baf427e30

**Comment by diervo** [comment]
After some digging, seems that is indeed purely missing as its on the official documentation: https://cloud.google.com/vertex-ai/docs/reference/rest/v1/Schema

**Comment by diervo** [comment]
I tested it and response_shema works without any code changes.
Would the team be interested if I do a PR fixing this?

**Comment by happy-qiao** [comment]
Hi diervo.

Could you send the PR to us? I just created types/common.ts file in https://github.com/googleapis/nodejs-vertexai/pull/404.
It has Schema interface. Could you help us adding responseSchema into GenerationConfig interface.

Thanks!

**Comment by philwhln** [comment]
This is supported in https://www.npmjs.com/package/@google/generative-ai . I thought I was on the wrong package, so switched over and found google auth doesn't work 🤦 , so need support here.

**Comment by diervo** [comment]
Will try to work on this over the weekend.

**Comment by happy-qiao** [comment]
Hi @diervo 

Have you started the implementation yet? Since I didn't receive a response from you for several days and I have a deadline to meet by Friday, I went ahead and created a PR to add this feature. https://github.com/googleapis/nodejs-vertexai/pull/414

**Comment by amit-pnkfrg** [comment]
Facing this as well :)

**Comment by happy-qiao** [comment]
We will have a release this week to add support for responseSchema. @amit-pnkfrg

**Comment by philwhln** [comment]
> I have a deadline to meet by Friday

@happy-qiao, just so you know, this only affects the TypeScript type checking. The functionality itself actually works. We were able to unblock by temporarily disabling type checks when we pass the the schema

```
// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-ignore
responseSchema: ...
```

**Comment by happy-qiao** [comment]
responseSchema added in https://github.com/googleapis/nodejs-vertexai/releases/tag/v1.6.0

**Comment by danielsigl** [comment]
@happy-qiao Thank you for adding this feature! Are there any plans to include this critical functionality in the documentation? It took me hours to figure this out, and I believe this is an essential feature that almost everyone would need to work effectively with the model output.
